File: game-server/src/main/java/org/forfun/mmorpg/game/admin/IpHandler.java
package org.forfun.mmorpg.game.admin;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.forfun.mmorpg.game.base.GameContext;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.servlet.HandlerInterceptor;

import java.util.Set;

public class IpHandler implements HandlerInterceptor {

    private Logger logger = LoggerFactory.getLogger(IpHandler.class);

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
            throws Exception {
        Set<String> whiteIps = GameContext.getServerConfig().getAdminIps();
        String realIp = IpUtil.getRealIp(request);
        boolean isSafe = whiteIps.contains(realIp);
        if (!isSafe) {
            logger.error("ip[{}]无法访问后台命令", realIp);
        }
        return isSafe;
    }
}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/admin/IpHandler.kt
package org.forfun.mmorpg.game.admin

import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.forfun.mmorpg.game.base.GameContext
import org.slf4j.LoggerFactory
import org.springframework.web.servlet.HandlerInterceptor

class IpHandler : HandlerInterceptor {

    private val logger = LoggerFactory.getLogger(IpHandler::class.java)

    override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
        val whiteIps = GameContext.serverConfig().adminWhiteIpsSet
        val realIp = IpUtil.getRealIp(request)
        val isSafe = whiteIps.contains(realIp)
        if (!isSafe) {
            logger.error("ip[$realIp]无法访问后台命令")
        }
        return isSafe
    }
}
